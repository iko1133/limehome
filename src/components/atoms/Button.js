import React from "react";
import styled from "styled-components/native";

import IconText from "./IconText";

const Container = styled.TouchableOpacity`
  background-color: ${({ theme }) => theme.BUTTON_BACKGROUND_COLOR};
  border-radius: 4px;
  justify-content: center;
  height: 60px;
  padding: 0 16px;
`;

const Text = styled(IconText)``;

const Button = ({ onPress, style, ...otherProps }) => {
  return (
    <Container onPress={onPress} style={style}>
      <Text {...otherProps} />
    </Container>
  );
};

export default Button;
