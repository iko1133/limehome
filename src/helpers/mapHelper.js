// Copies given object, except, switches names of the location property's properties for the
// marker objects to function properly
// lgn to longitude
// lat to latitude
export const fixObjectLocationProperty = (object) => ({
  ...object,
  location: {
    ...object.location,
    longitude: object.location?.lng,
    latitude: object.location?.lat,
  },
});
