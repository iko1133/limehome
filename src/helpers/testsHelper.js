import { ThemeProvider } from "styled-components/native";

import { lightTheme } from "styles/theme";

/**
 * Wraps given children into a ThemeProvider with lightTheme set as the main theme
 * @param {object} children - children to be rendered inside ThemeProvider
 */
export const WrapInThemeProvider = (children) => (
  <ThemeProvider theme={lightTheme}>{children}</ThemeProvider>
);
