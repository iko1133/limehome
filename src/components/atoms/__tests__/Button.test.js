import React from "react";
import renderer from "react-test-renderer";

import { WrapInThemeProvider } from "helpers/testsHelper";
import Button from "../Button";

// Fixing vector icons problem with jest
jest.mock(
  "@expo/vector-icons/build/vendor/react-native-vector-icons/lib/create-icon-set.js",
  () => {
    return () => "";
  }
);

describe("Testing Button", () => {
  it("Renders button with text 'Press' on it", () =>
    expect(
      renderer.create(WrapInThemeProvider(<Button text="Press" />))
    ).toMatchSnapshot());

  it("Renders button with text 'Press Again' on it with star icon", () =>
    expect(
      renderer.create(
        WrapInThemeProvider(<Button text="Press again" iconName="star" />)
      )
    ).toMatchSnapshot());

  it("Renders button with text 'Press Again' on it with star icon on the right", () =>
    expect(
      renderer.create(
        WrapInThemeProvider(
          <Button text="Press again" iconName="star" iconLeft={false} />
        )
      )
    ).toMatchSnapshot());
});
