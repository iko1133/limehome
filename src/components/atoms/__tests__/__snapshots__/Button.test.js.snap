// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Button Renders button with text 'Press Again' on it with star icon 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#4D6447",
      "borderRadius": 4,
      "height": 60,
      "justifyContent": "center",
      "opacity": 1,
      "paddingBottom": 0,
      "paddingLeft": 16,
      "paddingRight": 16,
      "paddingTop": 0,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        },
        Object {},
      ]
    }
  >
    <Text />
    <Text
      style={
        Array [
          Object {},
        ]
      }
    >
      Press again
    </Text>
  </View>
</View>
`;

exports[`Testing Button Renders button with text 'Press Again' on it with star icon on the right 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#4D6447",
      "borderRadius": 4,
      "height": 60,
      "justifyContent": "center",
      "opacity": 1,
      "paddingBottom": 0,
      "paddingLeft": 16,
      "paddingRight": 16,
      "paddingTop": 0,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        },
        Object {},
      ]
    }
  >
    <Text
      style={
        Array [
          Object {},
        ]
      }
    >
      Press again
    </Text>
    <Text />
  </View>
</View>
`;

exports[`Testing Button Renders button with text 'Press' on it 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#4D6447",
      "borderRadius": 4,
      "height": 60,
      "justifyContent": "center",
      "opacity": 1,
      "paddingBottom": 0,
      "paddingLeft": 16,
      "paddingRight": 16,
      "paddingTop": 0,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        },
        Object {},
      ]
    }
  >
    <Text />
    <Text
      style={
        Array [
          Object {},
        ]
      }
    >
      Press
    </Text>
  </View>
</View>
`;
